taskRequests/script.js

```diff
@@ -320,16 +366,55 @@ function createTaskRequestCard({ id, task, requestors, status, taskTitle }) {
   return card;
 }
 
-function renderTaskRequestCards(taskRequests) {
-  if (taskRequests.length > 0) {
-    filterContainer.classList.remove('hidden');
-    taskRequests.forEach((taskRequest) => {
+const intersectionObserver = new IntersectionObserver(async (entries) => {
+  if (!nextLink) {
+    return;
+  }
+  if (entries[0].isIntersecting && !isDataLoading) {
+    await renderTaskRequestCards({}, nextLink);
+  }
+});
+
+const addIntersectionObserver = () => {
+  intersectionObserver.observe(lastElementContainer);
+};
+const removeIntersectionObserver = () => {
+  intersectionObserver.unobserve(lastElementContainer);
+};
+
```

Comment:
can you give me some context of intersectionObserver. What is the purpose and why are we using it here. Thanks

Reply:
This is the web api which is useful to load more items in screen as the user scrolls down.
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API